### ISSUE TITLE
ValueError('signal only works in main thread')

### ISSUE THREAD

**Issue by maxwass** [PyTorchLightning/pytorch-lightning#9590]
### Discussed in https://github.com/PyTorchLightning/pytorch-lightning/discussions/9589

<div type='discussions-op-text'>

<sup>Originally posted by **maxwass** September 17, 2021</sup>
Has anyone else run into this error:
`ValueError('signal only works in main thread')`

I'm running a hyper parameter sweep using Weights and Biases's framework.

Running on a GPU on Google Colab which causes all launched runs to fail. Running it locally (Mac OS) prompts 'signal only works in main thread' to be printed to stdout (which also happens on Colab) but it doesn't crash.

Any ideas? It seems people using [Ray with PL](https://github.com/PyTorchLightning/pytorch-lightning/issues/3651) have come across this. The hacky solution presented there (`os.environ['SLURM_JOB_NAME'] = 'bash'`) doesn't work in my case (neither on Mac OS or Colab).</div>

**Comment by LukasHedegaard** [comment]
I'm experiencing the same issue. The standard lightning + ray example presented [here](https://docs.ray.io/en/latest/tune/tutorials/tune-pytorch-lightning.html) works as expected for pl v1.3 and v1.4, but fails for v1.5.
I'm running on a MacBook Pro (CPU only), so the work-around for SLURM jobs doesn't work here.

**Comment by paulhager** [comment]
I'm also experiencing this issue. I'm running on a linux machine with GPU (that is also being seen and activated by ray), so I don't think its a hardware issue. Seems to be the newest pl version has some incompatability

**Comment by vlievin** [comment]
Hi @maxwass,  am experiencing this issue as well with `ray[tune] + pytorch_lightning `. Setting `os.environ['SLURM_JOB_NAME'] = 'bash'` doesn't work. Downgrading `pytorch_lightning==1.4` seems to fix the problem.
 Is there a workaround for the newer versions of lightning?

**Comment by maxwass** [comment]
@vlievin unfortunately I'm not sure if anything other than downgrading works at this point. I was using wandb, not raytune, and fixed my problem by using all yaml files. See the discussion [here](https://github.com/PyTorchLightning/pytorch-lightning/discussions/9589).

**Comment by vlievin** [comment]
@maxwass thanks for the link. I see, it has been fixed in this [PR](https://github.com/PyTorchLightning/pytorch-lightning/pull/10610/files#diff-455634aa5542cb67cb1ec373b47aa36cea4a924164006add72afec20a14818c7). So I assume this problem will be fixed in the future releases. 

As of today, I could only install`pytorch_lightning==1.5.5`. In case someone stumbles on this problem with `pytorch_lightning<1.6`. I avoided the problem by patching the `Trainer`, so it avoids using `signal` completely using:
```python
def patch_signal_connector(trainer: Trainer):
    """
    Avoid using `signal` in `trainer.SignalConnector` if Lightning
    is not running in the main thread.
    """
    if threading.current_thread() is threading.main_thread():
        return

    warnings.warn(f"Lightning is not running in the main thread. "
                  f"Patching `trainer.SignalConnector` to disable `signal`.")

    def _no_signal_teardown(self):
        self._original_handlers = {}

    trainer.signal_connector._is_on_windows = lambda *_: True
    trainer.signal_connector.teardown = partial(_no_signal_teardown, trainer.signal_connector)
    
# usage
trainer = Trainer(...)
patch_signal_connector(trainer)
trainer.fit(...)
```

**Comment by denisoliveirac** [comment]
A regression was introduced in version 1.5.5 and the problem returned.
This [PR](https://github.com/PyTorchLightning/pytorch-lightning/pull/10610/files#diff-455634aa5542cb67cb1ec373b47aa36cea4a924164006add72afec20a14818c7) solved the problem. This code was released in version 1.5.3 and works also in 1.5.4, but in version 1.5.5 the problem returned.

**Comment by vlievin** [comment]
@denisoliveirac thanks for the info
